server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -1074,6 +1077,7 @@ record PluginServiceInstances(
             b.bind(SearchPhaseController.class).toInstance(new SearchPhaseController(searchService::aggReduceContextBuilder));
             b.bind(Transport.class).toInstance(transport);
             b.bind(TransportService.class).toInstance(transportService);
+            b.bind(NodeMetrics.class).toInstance(nodeMetrics);
```

Comment:
I'm not familiar with how Node startup works.  Are we sure that this has the same lifecycle as `nodeMetrics` , i.e. it starts/loads and stops/unloads at the same time?  If it doesn't, we might get exceptions for double registration.